### ISSUE TITLE
Enable Http2 Push Server - issue with admin page only

### ISSUE THREAD

**Issue by jeromeXoo** [cloudflare/Cloudflare-WordPress#204]
Hi,

The context: Nginx + php7fpm + Cloudflare **free plan**

I enabled the Http2 Push Server feature:
- For logged or not logged user, the front works correctly, the javascript and css files are well pushed.

- When I access the administration pages error 502,  I get a 502 error. In the nginx logs, I have "Can not Push Server (header size over 3072 Bytes)"

I tried to change the "fastcgi_buffer_size" and "fastcgi_buffer", and I modified the code of the cloudflare plugin to increase the limit "headersize":
`
if ($headerSize > 3072) {
                    error_log('Cannot Server Push (header size over 3072 Bytes).');
                    return $src;
                }
`
to
`
if ($headerSize > 7168) { {
                    error_log('Cannot Server Push (header size over 7168 Bytes).');
                    return $src;
                }
`
=> without success, header size were always too big (I do not understand why the headers are so big).
I looked for different solution by playing on the buffer, without success..

If I go back to my problem, the only pages that pose a problem are the administration pages, administration pages with push server enabled. I do not need to push server on these administration pages (and in general, I do not see any need).

If I disabled push server for administration pages:
```
    public static function initHooks()
    { 
		// Change: disable the push server if it is an admin page
		if (is_admin()) {
			return;
		};
	
        self::$initiated = true;

        add_action('wp_head', array('\CF\WordPress\HTTP2ServerPush', 'http2ResourceHints'), 99, 1);
        add_filter('script_loader_src', array('\CF\WordPress\HTTP2ServerPush', 'http2LinkPreloadHeader'), 99, 1);
        add_filter('style_loader_src', array('\CF\WordPress\HTTP2ServerPush', 'http2LinkPreloadHeader'), 99, 1);
    }
```
=> Everything is working fine, frontend is pushed, admin page without error and not pushed

Maybe there is a better solution? or can this solution be integrated without risk?

**Comment by thellimist** [comment]
The solution looks good. Feel free to open a PR. 

Thanks

**Comment by SolidSnakeLive** [comment]
I can confirm this issue... 
I'm using W3 total cache with cloudflare plugin and when I do purge cache this error shows up, although it was working fine before but now it gives error 500 in the browser
PHP message: Cannot Server Push (header size over 3072 Bytes)

**Comment by flexchar** [comment]
@blue928 or you can do the short `cloudflare.loader.php` edit as seen here: https://github.com/cloudflare/Cloudflare-WordPress/pull/214/commits/4a815fe2c9208bad1d7879d3c4e4029e94ec6e24. This will work until plugin is updated.

**Comment by sungkhum** [comment]
Same issue - thanks for this!

**Comment by LSB-Web-Design** [comment]
I'm seeing these errors in my apache error log. Tried the admin fix but still occurring when I clear my cache. Just wondering on the status of this issue?

**Comment by flexchar** [comment]
@LSB-Web-Design have you tried my fix?

**Comment by LSB-Web-Design** [comment]
Yes I've tried the wp admin pages fix. Seems to be occurring on non admin pages though.

**Comment by flexchar** [comment]
Oh, sounds like you might be having a different case. As such, I'd recommend opening a new issue.
My best guess would be that you might be loading too many resources on the frontend...

**Comment by agador** [comment]
I would also open a problem ticket with CloudFlare Support directly regarding the issue.  There are a number of outstanding issues with this plugin that they are "waiting on other issues" to even review some of the suggested fixes on this project.

Last I looked, they have not had any commits to this project in over a year now and the lack of PHP 7.3 support and other outstanding problems are not being addressed... as far as I can see.
